src/components/GuestInput.js

```diff
@@ -0,0 +1,135 @@
+import React from "react";
+import styled from "@emotion/styled";
+
+const StyledInputContainer = styled.div`
+  margin-bottom: 0.67em;
+  display: flex;
+  position: relative;
+
+  ${props =>
+    props.disabled &&
+    `
+    &::after {
+      position: absolute;
+      content: "MAX ${props.maxGuests} GUESTS";
+      top: 0;
+      bottom: 0;
+      left: 0;
+      right: 0;
+      font-size: 1.25rem;
+      background-color: rgba(250, 252, 254, 1);
+      border: 1px dashed rgba(220, 223, 226, 1);
+      border-radius: 0.25em;
+      text-align: center;
+      color: rgba(54, 63, 84, 0.4);
+      padding: 0 1em;
+      line-height: 2.7em;
+      transition: all 0.2s ease;
+      cursor: not-allowed;
+    }
+
+    &:hover::after {
+      color: rgba(54, 63, 84, 0.7);
+    }
+  `}
+`;
+
+const StyledLabel = styled.label`
+  position: relative;
+  width: fill-available;
```

Comment:
I believe Firefox categorizes `fill-available` as having poor cross-browser support. See the `fill` row in the support table:

https://developer.mozilla.org/en-US/docs/Web/CSS/width#Browser_compatibility

For Mozilla, it's `available`, not `fill-available`, so you'll need to add the `width: moz-available`.

https://css-tricks.com/almanac/properties/w/width/#article-header-id-11

---

Since it has poor browser support, maybe omit `width: fill-available` and add the following to the `StyledInputContainer` styles:

```css
display: grid;
grid-template-columns: 1fr auto;
```

See the example alternative here: https://jsfiddle.net/f08ay2bw/

I minimally reproduced your code, so if you uncomment and comment, you can see that both yield the same result, but grid has much better browser support.